File: app/Http/Livewire/Ticketing.php
<?php

namespace App\Http\Livewire;

use Illuminate\Support\Facades\Http;
use Illuminate\Support\Facades\Log;
use Livewire\Component;

class Ticketing extends Component
{
    public $subject;
    public $message;

    public function init(): void
    {
        $system = config('ticketing.platform');
        $response = null;

        switch ($system) {
            case 'zammad':
//                $response = $this->zammad();
                break;
            case 'osticket':
                $this->osticket($this->subject, $this->message);
                break;
        }
    }

    // Submit a ticket using the zammad API library
//    public function zammad($title, $description): \Response
//    {
//
//    }

    // Submit a ticket using the osTicket API
    public function osticket($title, $description): void
    {
        $base_url = config('ticketing.osticket_base_url');

        $url = $base_url.'/api/tickets.json';

        $headers = [
            'X-API-Key' => config('ticketing.osticket_api_key'),
        ];

        $response = Http::withHeaders($headers)->post($url, [
            'alert' => true,
            'source' => 'API',
            'name' => auth()->user()->name,
            'email' => auth()->user()->email,
            'subject' => $title,
            'message' => 'data:text/plain,'.$description
        ]);

        if ($response->successful()) {
            session()->flash('success', 'Issue reported successfully!');
        } else {
            Log::error('Error reporting issue to osTicket: '.$response->body());
            session()->flash('error', 'There was an error reporting your issue. Please try again.');
        }
    }


    /**
     * Get the view / contents that represent the component.
     *
     * @return \Illuminate\View\View|string
     */
    public function render()
    {
        return view('livewire.ticketing');
    }
}

File: pkg/ticketing/ticketing.go
package ticketing

import (
	"bytes"
	"encoding/json"
	"errors"
	"intralab/config"
	"net/http"
)

type Ticket struct {
	Name    string `json:"name"`
	Email   string `json:"email"`
	Subject string `json:"subject"`
	Message string `json:"message"`
}

func OSTicket(cfg config.TicketingConfig, ticket Ticket) error {
	url := cfg.OSTicket.BaseURL + "/api/tickets.json"
	apiKey := cfg.OSTicket.APIKey

	headers := http.Header{}
	headers.Set("X-API-Key", apiKey)

	data := map[string]string{
		"alert":   "true",
		"source":  "API",
		"name":    ticket.Name,
		"email":   ticket.Email,
		"subject": ticket.Subject,
		"message": ticket.Message,
	}

	response, err := postRequest(url, data, headers)
	if err != nil {
		return err
	}
	defer response.Body.Close()

	if response.StatusCode != 201 {
		var results map[string]interface{}
		err := json.NewDecoder(response.Body).Decode(&results)
		if err != nil {
			return err
		}

		return errors.New(results["error"].(string))
	}
	return nil
}

func postRequest(url string, data map[string]string, headers http.Header) (*http.Response, error) {
	jsonData, err := json.Marshal(data)
	if err != nil {
		return nil, err
	}

	request, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonData))
	if err != nil {
		return nil, err
	}

	request.Header = headers

	client := &http.Client{}
	response, err := client.Do(request)
	if err != nil {
		return nil, err
	}

	return response, nil
}
